tests/tensor_search/test_search.py

```diff
@@ -54,6 +56,44 @@ def test_each_doc_returned_once(self):
         )
         assert len(search_res['hits']) == 2
 
+    @mock.patch.dict('os.environ', {**os.environ, **{'MARQO_MAX_SEARCHABLE_TENSOR_ATTRIBUTES': '2'}})
+    def test_search_with_excessive_searchable_attributes(self):
+        with self.assertRaises(InvalidArgError):
+            tensor_search.add_documents(
+                config=self.config, index_name=self.index_name_1, docs=[
+                    {"abc": "Exact match hehehe", "other field": "baaadd", "_id": "5678"},
+                    {"abc": "random text", "other field": "Close match hehehe", "_id": "1234"},
+                ], auto_refresh=True)
+            tensor_search.search(
+                config=self.config, index_name=self.index_name_1, text="Exact match hehehe",
+                searchable_attributes=["abc", "def", "other field"], return_doc_ids=True
+            )
+
+    @mock.patch.dict('os.environ', {**os.environ, **{'MARQO_MAX_SEARCHABLE_TENSOR_ATTRIBUTES': '0'}})
+    def test_search_with_excessive_searchable_attributes(self):
+        with self.assertRaises(InvalidArgError):
+            tensor_search.add_documents(
+                config=self.config, index_name=self.index_name_1, docs=[
+                    {"abc": "Exact match hehehe", "other field": "baaadd", "_id": "5678"},
+                    {"abc": "random text", "other field": "Close match hehehe", "_id": "1234"},
+                ], auto_refresh=True)
+            tensor_search.search(
+                config=self.config, index_name=self.index_name_1, text="Exact match hehehe",
+                searchable_attributes=["other field"], return_doc_ids=True
+            )
+
+    @mock.patch.dict('os.environ', {**os.environ, **{'MARQO_MAX_SEARCHABLE_TENSOR_ATTRIBUTES': '2'}})
+    def test_search_with_no_searchable_attributes_but_max_searchable_attributes_env_set(self):
+        with self.assertRaises(InvalidArgError):
+            tensor_search.add_documents(
+                config=self.config, index_name=self.index_name_1, docs=[
+                    {"abc": "Exact match hehehe", "other field": "baaadd", "_id": "5678"},
+                    {"abc": "random text", "other field": "Close match hehehe", "_id": "1234"},
+                ], auto_refresh=True)
+            tensor_search.search(
+                config=self.config, index_name=self.index_name_1, text="Exact match hehehe", return_doc_ids=True
+            )
+
```

Comment:
How about a test with a really large `MARQO_MAX_SEARCHABLE_TENSOR_ATTRIBUTES`, and a low number of document fields. Then assert it should fail if it searchable attributes isn't set. 

This ensures we're testing the codepath where it should fail if searchable attributes isn't set

Reply:
removing `field_names` will solve this problem

Reply:
what do you mean?

Reply:
We were talking about different issues. Test `test_search_with_allowable_num_searchable_attributes` Should resolve this.

Reply:
I think we still need this test. We want to ensure an error is raised if `MARQO_MAX_SEARCHABLE_TENSOR_ATTRIBUTES` is set but `searchable_attributes` aren't included in `search()`. 

We need to ensure that errors are raised due to `searchable_attributes` not being included in `search()` rather than the `len(searchable_attributes) > MARQO_MAX_SEARCHABLE_TENSOR_ATTRIBUTES`

In this particular test it is ambiguous why the error is raised, because the total number of implicit searchable attributes (3) is larger than `MARQO_MAX_SEARCHABLE_TENSOR_ATTRIBUTES` (2)

Reply:
We have those tests in `TestValidateSearchableAttributes` but we can also just have them here.

Reply:
`test_search_with_no_searchable_attributes_but_max_searchable_attributes_env_set` is that test? Or are you just wanting `MARQO_MAX_SEARCHABLE_TENSOR_ATTRIBUTES > len(fields)`?

Reply:
Both cases:
`MARQO_MAX_SEARCHABLE_TENSOR_ATTRIBUTES > len(fields)`
and 
`MARQO_MAX_SEARCHABLE_TENSOR_ATTRIBUTES < len(fields)`